packages/api/src/controllers/user.js

```diff
@@ -403,4 +434,247 @@ app.post(
   },
 )
 
+app.post(
+  '/create-customer',
+  validatePost('create-customer'),
+  async (req, res) => {
+    const { data: userIds } = await req.store.query({
+      kind: 'user',
+      query: { email: req.body.email },
+    })
+    if (userIds.length < 1) {
+      res.status(404)
+      return res.json({ errors: ['user not found'] })
+    }
+
+    let user = await req.store.get(`user/${userIds[0]}`, false)
+    if (user) {
+      const customer = await stripe.customers.create({
+        email: req.body.email,
+      })
+      user = { ...user, stripeCustomerId: customer.id }
+      await req.store.replace(user)
+      res.status(201)
+      res.json(customer)
+    } else {
+      res.status(403)
+      res.json({ errors: ['customer not created'] })
+    }
+  },
+)
+
+app.post(
```

Comment:
Seems like we'd want authMiddleware for a specific user or admin here

Reply:
Does that middleware already exist? What would that look like?